scylla-server-rust/src/main.rs

```diff
@@ -0,0 +1,41 @@
+use std::sync::Arc;
+
+use axum::{http::Method, routing::get, Router};
+use scylla_server_rust::{
+    controller::{
+        self, data_type_controller, driver_controller, location_controller, node_controller,
+        run_controller, system_controller,
+    },
+    prisma::PrismaClient,
+    Database,
+};
+use tower_http::cors::{Any, CorsLayer};
+
+#[tokio::main]
+async fn main() {
+    let client: Database = Arc::new(PrismaClient::_builder().build().await.unwrap());
+
+    let app = Router::new()
+        .route(
+            "/data/:dataTypeName/:runId",
+            get(controller::data_controller::get_data),
+        )
+        .route("/datatypes", get(data_type_controller::get_all_data_types))
+        .route("/drivers", get(driver_controller::get_all_drivers))
+        .route("/locations", get(location_controller::get_all_locations))
+        .route("/nodes", get(node_controller::get_all_nodes))
+        .route("/runs", get(run_controller::get_all_runs))
+        .route("/runs/:id", get(run_controller::get_run_by_id))
+        .route("/systems", get(system_controller::get_all_systems))
+        .layer(
+            CorsLayer::new()
+                // allow `GET`
+                .allow_methods([Method::GET])
+                // allow requests from any origin
+                .allow_origin(Any),
+        )
+        .with_state(client);
```

Comment:
is this the suggested way of doing this? is it not possible to just have a global we just import into each of the service functions, I would get it if rust doesnt like that though. However it makes me uncomfy to have the prisma client inside the controllers when they really dont need it

Reply:
See my above comment on passing `Database` in.  Its related to async and tokio stuff.  I get you are worried about it being where it doesnt need to be, but I cant think of a different way thats also thread safe, load balancing safe, etc.